lib/src/over_react_redux/hooks/use_selector.dart

```diff
@@ -0,0 +1,183 @@
+/// customize how equality is defined, pass a comparator function to the [equalityFn] argument.
+///
+/// If you need to use a selector with custom [Context], use [createSelectorHook] instead.
+///
+/// See the [react-redux JS documentation](https://react-redux.js.org/api/hooks#useselector) for more details.
+///
+/// ### Example
+///
+/// > This example assumes that your `Counter` component is rendered as the descendant of a [ReduxProvider] component
+///   that is wired up to a Redux `Store` instance with a `CounterState` instance containing a field called `count`.
+///
+/// ```dart
+/// import 'package:over_react/over_react.dart';
+/// import 'package:over_react/over_react_redux.dart';
+/// import 'counter_state.dart';
+///
+/// mixin CounterProps on UiProps {}
+///
+/// UiFactory<CounterProps> Counter = uiFunction(
+///   (props) {
+///     final count = useSelector<int, CounterState>((state) => state.count);
+///
+///     return Dom.div()('The current count is $count');
+///   },
+///   $CounterConfig, // ignore: undefined_identifier
+/// );
+/// ```
+///
+/// > Related: [useDispatch]
+TSelector useSelector<TSelector, TReduxState>(TSelector Function(TReduxState state) selector, [
+  bool Function(TSelector tNextSelector, TSelector tPrevSelector) equalityFn,
+]) {
+  ReactInteropValue jsSelector(ReactInteropValue jsState) => wrapInteropValue(selector(unwrapInteropValue(jsState)));
+  _JsReduxStateEqualityFn jsEqualityFn = equalityFn == null
+      ? null
+      : allowInterop((nextJsSelector, prevJsSelector) =>
+          equalityFn(unwrapInteropValue(nextJsSelector), unwrapInteropValue(prevJsSelector)));
+
+  return unwrapInteropValue<TSelector>(_jsUseSelector(allowInterop(jsSelector), jsEqualityFn));
+}
+
+@JS('ReactRedux.useSelector')
+external ReactInteropValue _jsUseSelector(_JsSelectorFn selector, [_JsReduxStateEqualityFn equalityFn]);
+
+// ----------------------------------------------------
+//  createSelectorHook
+// ----------------------------------------------------
+
+/// Returns a function that can be named and used as a replacement for [useSelector] when you want to
+/// use custom [context].
+///
+/// This is useful if you're building a complex reusable component, and you don't want your `Store` to collide
+/// with any Redux `Store` your consumers' applications might use.
+///
+/// See the [react-redux JS documentation](https://react-redux.js.org/api/hooks#custom-context) for more details.
+///
+/// ### Custom Context Example
+///
+/// ```dart
+/// import 'package:over_react/over_react.dart';
+/// import 'package:over_react/over_react_redux.dart';
+/// import 'package:redux/redux.dart';
+///
+/// // ------------------------------------
+/// //  1. Declare the custom context
+/// // ------------------------------------
+/// final MyContext = createContext<MyState>();
+/// final useMySelector = createSelectorHook(MyContext);
```

Comment:
I wonder if we should should recommend shadowing the top-level useSelector/useStore/etc hooks in our doc comments and examples, for consistency with what they show in React Redux https://react-redux.js.org/api/hooks#custom-context